fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/temp/properties/PredicateHeightProperty.java

```diff
@@ -42,30 +42,30 @@
     private static final PredicateHeightProperty INSTANCE = new PredicateHeightProperty();
 
     @Override
-    public boolean shouldVisit(@Nonnull PlannerExpression expression) {
+    public boolean shouldVisit(@Nonnull RelationalExpression expression) {
         return true;
     }
 
     @Override
-    public boolean shouldVisit(@Nonnull ExpressionRef<? extends PlannerExpression> ref) {
+    public boolean shouldVisit(@Nonnull ExpressionRef<? extends RelationalExpression> ref) {
         return true;
     }
 
     @Nonnull
     @Override
-    public Integer evaluateAtExpression(@Nonnull PlannerExpression expression, @Nonnull List<Integer> childResults) {
+    public Integer evaluateAtExpression(@Nonnull RelationalExpression expression, @Nonnull List<Integer> childResults) {
         int maxChildDepth = 0;
         for (Integer childDepth : childResults) {
             if (childDepth != null && childDepth > maxChildDepth) {
                 maxChildDepth = childDepth;
             }
         }
-        return maxChildDepth + (expression instanceof QueryPredicate ? 1 : 0);
```

Comment:
This seems to contradict with the purpose of the property as stated in the java doc of the class. Are you counting `QueryComponent` or `QueryPredicate`?

Reply:
Ah, good point. If we wanted to implement this, we couldn't use the infrastructure for `PlannerProperty` anymore.

On the other hand, it looks like this is dead code, so I've deleted it instead.